wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheMonitor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.cache;
+
+/**
+ * Interface for cache health monitors. This class uses a background thread to monitor cache
+ * cluster health and manage state transitions between HEALTHY, SUSPECT, and DEGRADED states.
+ */
+public interface CacheMonitor extends Runnable {
+
+  /**
+   * Registers a cache cluster for monitoring.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @param connection the cache connection for executing health checks
+   */
+  void registerCluster(String endpoint, CacheConnection connection);
+
+  /**
+   * Gets the current health state of a cache cluster.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @return the current state (HEALTHY, SUSPECT, or DEGRADED)
+   */
+  ClusterState getClusterState(String endpoint);
+
+  /**
+   * Reports a cache operation error to trigger state transitions.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @param error the exception that occurred during the cache operation
+   */
+  void reportError(String endpoint, Exception error, String operation);
+
+  /**
+   * Increments the in-flight write size for a cache cluster.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @param bytes the size of the write operation in bytes (key hash + value)
+   */
+  void incrementInFlightSize(String endpoint, long bytes);
+
+  /**
+   * Decrements the in-flight write size for a cache cluster.
+   *
+   * @param endpoint the cache cluster endpoint (hostname:port)
+   * @param bytes the size of the write operation in bytes (key hash + value)
+   */
+  void decrementInFlightSize(String endpoint, long bytes);
+
+  /**
+   * Whether this {@link CacheMonitor} has stopped monitoring.
+   *
+   * @return true if the monitoring has stopped; false otherwise.
+   */
+  boolean isStopped();
+
+  /**
+   * Gets the current in-flight write size for an endpoint.
+   * For debugging purposes.
+   */
+  long getInFlightSize(String endpoint);
+
+  /**
+   * Enum representing the health state of a cache cluster.
+   */
+  enum ClusterState {
```

Comment:
Rename to `HealthState` because this can represent both endpoint state and entire cluster state.